File: android/src/main/kotlin/com/example/notification_voip_plugin/models/CallState.kt
package com.example.notification_voip_plugin.models

data class CallState(
    val callId: String,
    var status: String = "ringing", // ringing, connected, onHold, ended
    var isMuted: Boolean = false,
    var isSpeakerOn: Boolean = false,
    var isCameraOn: Boolean = false,
    var connectedAt: Long? = null
) {
    fun toMap(): Map<String, Any?> = mutableMapOf<String, Any?>(
        "callId" to callId,
        "status" to status,
        "isMuted" to isMuted,
        "isSpeakerOn" to isSpeakerOn,
        "isCameraOn" to isCameraOn
    ).also {
        if (connectedAt != null) it["connectedAt"] = connectedAt
    }
}

File: ios/notification_voip_plugin/Sources/notification_voip_plugin/Models/CallState.swift
import Foundation

struct CallState {
    let callId: String
    var status: String  // ringing, connected, onHold, ended
    var isMuted: Bool = false
    var isSpeakerOn: Bool = false
    var isCameraOn: Bool = false
    var connectedAt: Date? = nil

    func toMap() -> [String: Any?] {
        return [
            "callId": callId,
            "status": status,
            "isMuted": isMuted,
            "isSpeakerOn": isSpeakerOn,
            "isCameraOn": isCameraOn,
            "connectedAt": connectedAt.map { Int($0.timeIntervalSince1970 * 1000) }
        ]
    }
}
